### PR TITLE
Remove unnecessary defines from half.hpp

### PR DESCRIPTION
**Pull request by JanuszL**
- removes defines for target from half.hpp as
  they are not part of the device code and host compiler
  evaluates it to false anyway

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- removes defines for target from half.hpp as
  they are not part of the device code and host compiler
  evaluates it to false anyway
- related to https://github.com/NVIDIA/DALI/issues/5651

### Affected modules and functionalities:
- half.hpp

### Key points relevant for the review:
- NA

### Tests:

- [x] Existing tests apply
  - compilation process and all DALI tests 
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: include/dali/util/half.hpp
Comment (mzient) [inline include/dali/util/half.hpp:1]: So, this file was unusable when compiling for CUDA, right? I think we should keep it this way and add:
```C++
#ifdef __CUDA_ARCH__
#error This header is not intended for use in CUDA device code.
#endif
```

**Comment by JanuszL** [comment]
Based on the offline discussion, as this header is not intended to be used in the device code there is no value in polishing it.

**Pull request closed without merging**